chore: use dependabot groups

matifali opened the pull request:
See https://github.blog/changelog/2023-06-30-grouped-version-updates-for-dependabot-public-beta/ for details. This will reduce the number of dependent PRs

For go and node dependencies, I leave the exact grouping to engineers who better know how they should be grouped.

matifali committed:
chore: use dependabot groups

See https://github.blog/changelog/2023-06-30-grouped-version-updates-for-dependabot-public-beta/ for details
This will reduce the number of dependabot PRs

For go and node dependencies, I leave the exact grouping to engineers who better know which one and how should be grouped.

matifali committed:
make fmt

coadler reviewed (changes requested):
I don't really think this is how we should use groups. We want separate commits for each dependency so we can debug them individually, and properly bisect them if they cause a regression.

One thing that I can think of that would benefit from this is our otel deps, which all need to be updated in tandem.
```
	go.nhat.io/otelsql v0.9.0
	go.opentelemetry.io/otel v1.14.0
	go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.14.0
	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.14.0
	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp v1.14.0
	go.opentelemetry.io/otel/sdk v1.14.0
	go.opentelemetry.io/otel/trace v1.14.0
```

ammario commented:
If we came across a large dependabot upgrade commit we could still continue the log2(n) bisect, albeit with some manual effort.

matifali committed:
group only otel for go remove grouping for node

matifali commented:
I removed the super groups for both node and go and added a group for `otel` only
@coadler

coadler commented on .github/dependabot.yaml:
```diff
@@ -110,5 +129,7 @@ updates:
       prefix: "chore"
     labels: []
-    ignore:
-      - dependency-name: "coder/coder"
```
Why was this removed?

matifali replied:
Because IMO for dogfood we should always have the latest terraform provider. For example templates, we are ignoring and prefer manual updates.

coadler replied:
Normally changes to our Terraform provider come with added features, changes, or deprecations that need to be added to the example templates as well. I think it makes sense to do these changes at the same time.

matifali replied:
Sure. I agree. I will revert this.

coadler reviewed (commented):
Looks like you removed it for GHA and not node

matifali committed:
better name groups and remove npm group

matifali committed:
add back empty labels

matifali committed:
ignore coder/coder for dogfood

Also removing the group for dogfood-docker as we only have a docker provider so nothing really to group here.

matifali committed:
make fmt
